disable clipboard history

SnowzNZ opened the pull request:
#247

not sure if this is how you would call a function and also use code

would also be good if someone could confirm the default state of the service

undergroundwires commented:
Awesome, thank you @SnowzNZ . I'm happy to receive first PR utilizing more complex logic with functions, these kind of PRs make my job much easier so I can delegate these stuff and focus on some nice features like import/export etc 🥹❤️

1. Thank you for pointing out double-checking default state, I had W11 21H2 VM with fresh installation when I saw the PR and see that:

-  Default startup mode for services are `Automatic` not `Manual`.
- I see that `HKLM\SOFTWARE\Policies\Microsoft\Windows\System\AllowClipboardHistory`  key does not exist on clean installation. So on revert we should delete the key with `reg delete` instead of adding `1`. 

2. Syntax, after `call:`, create an array and combine like this:

```yaml
call:
  - 
     function: DisablePerUserService
     // ..
  - 
     function: RunInlineCode
     parameters:
          code: reg add ...
          revertCode: reg add..
```

This way it will be sequence of calls and compiler will handle both. After this syntax change integration and unit tests will be green so it will not break.

SnowzNZ commented:
> Default startup mode for services are `Automatic` not `Manual`.

Changed. I just took the value that the user in the issue used.

> I see that HKLM\SOFTWARE\Policies\Microsoft\Windows\System\AllowClipboardHistory key does not exist on clean installation. So on revert we should delete the key with reg delete instead of adding 1.

Changed.

> Syntax, after call:, create an array and combine like this: ...

Fixed.

undergroundwires commented:
Yea, we did not conduct a review of revert logic there. All looks good @SnowzNZ, I will merge this in next minor release.

undergroundwires commented:
Very simple conflict. I cannot solve this, I will add this manually today.
